.template/addons/github/.github/workflows/test.yml.tt

```diff
@@ -50,18 +50,43 @@ jobs:
       - name: Push Docker image
         run: docker-compose push test
 
+  linting:
+    name: Linting
+    runs-on: ubuntu-latest
+    needs: build
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env BRANCH_TAG
+        uses: nimblehq/branch-tag-action@v1.2
+
+      - name: Login to Docker registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.DOCKER_REGISTRY_HOST }}
+          username: ${{ env.DOCKER_REGISTRY_USERNAME }}
+          password: ${{ env.DOCKER_REGISTRY_TOKEN }}
+
+      - name: Pull Docker image
+        run: docker-compose pull test || true
+
+      - name: Run codebase test
+        run: docker-compose run test bundle exec rspec spec/codebase/codebase_spec.rb --format progress
+
   unit_tests:
     name: Unit tests
     runs-on: ubuntu-latest
-    needs: build
+    needs: linting
```

Comment:
Should we isolate 3 jobs? As we still need to run the Unit tests even if the linting fails.
And in other projects, we also have separated lint, and test workflows.
```suggestion
    needs: build
```

Reply:
> Should we isolate 3 jobs? As we still need to run the Unit tests even if the linting fails.

For me, I don't think we need to run the unit test there as we have to fix the codebase issue first

> And in other projects, we also have separated lint, and test workflows.

Hmm, I'm not sure about that idea, it mean we will need to move the `build` step to another workflow 🤔 I would prefer a separate PR if you prefer that way @hoangmirs and might I know which project we have separate `linting and test` workflow @hoangmirs

Reply:
That's a good question 🤔 

IMO it depends what we want to favor:
a. reduce CI/CD run time? Then we better not run tests when linting is failing (more efficient)
b. unit/system tests results at the earliest? This might save 2 or 3 minutes to get the Unit Tests results as they will start BEFORE the end of the Lint workflow. (more effective)

I think we might prefer `b` (i.e. Hoang's suggestion).

Reply:
In general I think we will go with the option `a`, @hoangmirs as a Web chapter lead, could you bring this topic into the Web chapter to discuss

Reply:
yeah, updated for both `unit test` and `system test` @malparty @hoangmirs

Reply:
Good for me 👍 
If you have time to bring it into Flexi :D :D :D

Reply:
You would like to have that on Flexi too? Will do it in on of my PR 🤞

Reply:
Adjustmend Flexi on the PR 583 😄